virtual_rainforest/core/data.py

```diff
@@ -334,19 +335,92 @@ def load_data_config(self, data_config: dict[str, Any]) -> None:
         if not clean_load:
             raise ConfigurationError("Data configuration did not load cleanly")
 
-    def save_to_netcdf(self, output_file_path: Path) -> None:
-        """Save the entire contents of the data object as a NetCDF file.
+    def save_to_netcdf(
+        self, output_file_path: Path, variables_to_save: Optional[list[str]] = None
+    ) -> None:
+        """Save the contents of the data object as a NetCDF file.
+
+        Either the whole contents of the data object or specific variables of interest
+        can be saved using this function.
 
         Args:
             output_file_path: Path location to save the Virtual Rainforest model state.
+            variables_to_save: List of variables to be saved. If not provided then all
+                variables are saved.
         """
 
         # Check that the folder to save to exists and that there isn't already a file
         # saved there
         check_outfile(output_file_path)
 
-        # If the file path is okay then write the model state out as a NetCDF
-        self.data.to_netcdf(output_file_path)
+        # If the file path is okay then write the model state out as a NetCDF. Should
+        # check if all variables should be saved or just the requested ones.
+        if variables_to_save:
+            self.data[variables_to_save].to_netcdf(output_file_path)
+        else:
+            self.data.to_netcdf(output_file_path)
+
+    def append_to_netcdf(
```

Comment:
I think this comment goes along the lines already discussed, but having to load the existing dataset, appending new data and saving it again is probably not that efficient but, specially, rather cumbersome.

I'd simplify things as much as possible and save each step in its own file and then, at the very end of the run, compile the data in a single file or in chunks. The fewer IO operations you have during the execution of the calculation, the better.